google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -584,30 +780,92 @@ public Database getDatabase(String databaseName) throws SpannerException {
 
   @Override
   public OperationFuture<Backup, CreateBackupMetadata> createBackup(
-      String instanceName, String backupId, Backup backup) throws SpannerException {
-    acquireAdministrativeRequestsRateLimiter();
+      final String instanceName, final String backupId, final Backup backup)
+      throws SpannerException {
     CreateBackupRequest request =
         CreateBackupRequest.newBuilder()
             .setParent(instanceName)
             .setBackupId(backupId)
             .setBackup(backup)
             .build();
-    GrpcCallContext context = newCallContext(null, instanceName);
-    return databaseAdminStub.createBackupOperationCallable().futureCall(request, context);
+    OperationFutureCallable<CreateBackupRequest, Backup, CreateBackupMetadata> callable =
+        new OperationFutureCallable<CreateBackupRequest, Backup, CreateBackupMetadata>(
+            CreateBackupMetadata.class,
+            databaseAdminStub.createBackupOperationCallable(),
+            request,
+            instanceName,
+            new OperationsLister() {
+              @Override
+              public Paginated<Operation> listOperations(String nextPageToken) {
+                return listBackupOperations(
+                    instanceName,
+                    0,
+                    String.format(
+                        "(metadata.name:%s) AND (metadata.@type:type.googleapis.com/%s)",
+                        String.format("%s/backups/%s", instanceName, backupId),
+                        CreateBackupMetadata.getDescriptor().getFullName()),
+                    nextPageToken);
+              }
+            },
+            new Function<CreateBackupMetadata, Timestamp>() {
+              @Override
+              public Timestamp apply(CreateBackupMetadata input) {
+                return input.getProgress().getStartTime();
+              }
+            });
+    return RetryHelper.runWithRetries(
+        callable,
+        databaseAdminStubSettings
+            .createBackupOperationSettings()
+            .getInitialCallSettings()
+            .getRetrySettings(),
```

Comment:
I think we don't specify retry settings for non-idempotent RPCs. Do you know what this will default to then?

Reply:
There are defaults defined in the gapic configuration file as well, but those are already overridden by the client library. The defaults are set in the constructor of `SpannerOptions.Builder`:

https://github.com/googleapis/java-spanner/blob/9e5a1cdaacf71147b67681861f063c3276705f44/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java#L242-L295

Reply:
So does this mean that the Java client library has its own retrying settings which are different from the gapic config file?